cardinal/cardinal.go

```diff
@@ -0,0 +1,347 @@
+package cardinal
+
+import (
+	"errors"
+	"github.com/rotisserie/eris"
+	"pkg.world.dev/world-engine/cardinal/iterators"
+	"pkg.world.dev/world-engine/cardinal/storage/redis"
+	"pkg.world.dev/world-engine/cardinal/system"
+	"pkg.world.dev/world-engine/cardinal/types"
+	"pkg.world.dev/world-engine/cardinal/types/engine"
+	"strconv"
+)
+
+var (
+	ErrEntityMutationOnReadOnly          = errors.New("cannot modify state with read only context")
+	ErrEntityDoesNotExist                = iterators.ErrEntityDoesNotExist
+	ErrEntityMustHaveAtLeastOneComponent = iterators.ErrEntityMustHaveAtLeastOneComponent
+	ErrComponentNotOnEntity              = iterators.ErrComponentNotOnEntity
+	ErrComponentAlreadyOnEntity          = iterators.ErrComponentAlreadyOnEntity
+	ErrComponentNotRegistered            = iterators.ErrMustRegisterComponent
```

Comment:
any reason for aliasing the errors? why not just use them directly?

Reply:
mostly devex reason, we wanted to make all functions, errors, etc that is relevant to game devs available in `cardinal`.